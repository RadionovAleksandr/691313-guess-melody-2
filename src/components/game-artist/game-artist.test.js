import React from 'react';
import renderer from 'react-test-renderer';
import GameArtist from './game-artist.jsx';

it(`renders correctly GameGenre`, () => {
  const tree = renderer.create(
      <GameArtist
        question={{
          type: `artist`,
          song: {
            artist: `Jim Beam`,
            src: `https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg`,
          },
          answers: [
            {
              picture: `http://placehold.it/134x134`,
              artist: `John Snow`,
            },
            {
              picture: `http://placehold.it/134x134`,
              artist: `Jack Daniels`,
            },
            {
              picture: `http://placehold.it/134x134`,
              artist: `Jim Beam`,
            },
          ],
        }}
        onAnswer={() => {}}
      />).toJSON();

  expect(tree).toMatchSnapshot();
});
