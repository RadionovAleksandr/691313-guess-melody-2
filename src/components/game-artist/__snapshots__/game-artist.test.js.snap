// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly GameGenre 1`] = `
<section
  className="game game--artist"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <div
      className="timer__value"
      xmlns="http://www.w3.org/1999/xhtml"
    >
      <span
        className="timer__mins"
      >
        05
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        00
      </span>
    </div>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Кто исполняет эту песню?
    </h2>
    <div
      className="game__track"
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          type="button"
        />
        <div
          className="track__status"
        >
          <audio
            src="https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg"
          />
        </div>
      </div>
    </div>
    <form
      className="game__artist"
      onChange={[Function]}
    >
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="artist-0"
          name="answer"
          type="radio"
          value="artist-0"
        />
        <label
          className="artist__name"
          htmlFor="artist-0"
        >
          <img
            alt="John Snow"
            className="artist__picture"
            src="http://placehold.it/134x134"
          />
          John Snow
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="artist-1"
          name="answer"
          type="radio"
          value="artist-1"
        />
        <label
          className="artist__name"
          htmlFor="artist-1"
        >
          <img
            alt="Jack Daniels"
            className="artist__picture"
            src="http://placehold.it/134x134"
          />
          Jack Daniels
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="artist-2"
          name="answer"
          type="radio"
          value="artist-2"
        />
        <label
          className="artist__name"
          htmlFor="artist-2"
        >
          <img
            alt="Jim Beam"
            className="artist__picture"
            src="http://placehold.it/134x134"
          />
          Jim Beam
        </label>
      </div>
    </form>
  </section>
</section>
`;
